### PR TITLE
Ft setup menu endpoint 156978740

### PR DESCRIPTION
**Pull request by samuelbwambale**
PT story: https://www.pivotaltracker.com/n/projects/2166953/stories/156978740

### CODE REVIEW

File: app/api/menus_resource.py [hunk 1]
```diff
@@ -1,73 +1,36 @@
...
-         "description": 1003
-     }]}
-]
+parser = reqparse.RequestParser()
+parser.add_argument('forToday', type= bool, required=True, location="json")
+parser.add_argument('meals', type=list, required=True, location="json")
 
 
 class MenuResource(Resource):
     def post(self):
```
Comment (malmike) [inline app/api/menus_resource.py:13]: Am expecting to see doc strings. Try to use a linter for example pylint or PEP8

File: app/api/menus_resource.py [hunk 2]
```diff
@@ -1,73 +1,36 @@
...
-            "description": 1003
-        }])
-        menu.addMenu()
-        return jsonify({
-            'menus': menus }), 200
+        data = parser.parse_args()        
+        new_menu = Menu(
+        forToday = data['forToday'],
+        meals = data['meals']) 
+        menus.append(new_menu)
```
Comment (malmike) [inline app/api/menus_resource.py:18]: You should ensure to return status codes

File: app/api/menus_resource.py [hunk 3]
```diff
@@ -1,73 +1,36 @@
...
-            {
-            "price": "something soemthing",
-            "id": 1,
-            "description": 1003
-        }])
-        menu.addMenu()
-        return jsonify({
-            'menus': menus }), 200
+        data = parser.parse_args()        
+        new_menu = Menu(
```
Comment (malmike) [inline app/api/menus_resource.py:15]: Validate and verify the data that is passed. Basing off this structure the menu is always created even when there is unclear data passed

File: app/api/menus_resource.py [hunk 4]
```diff
@@ -1,73 +1,36 @@
...
-        },
-            {
-            "price": "something soemthing",
-            "id": 1,
-            "description": 1003
-        }])
-        menu.addMenu()
-        return jsonify({
-            'menus': menus }), 200
+        data = parser.parse_args()
```
Comment (malmike) [inline app/api/menus_resource.py:14]: What would happen if I plan to add a meal to an existing menu?

File: app/api/menus_resource.py [hunk 5]
```diff
@@ -1,73 +1,36 @@
...
+
+    def get(self):
+        data = parser.parse_args()
+        my_menu = Menu.get_menu_by_id(data['id'])
+        if not my_menu:
+            return {'message': 'Menu does not exist'. format(data['id'])}
+        
+        if data['forToday'] == True:
+            return jsonify({'my_menu': my_menu}), 200
+        else:
```
Comment (malmike) [inline app/api/menus_resource.py:30]: Status code?

File: app/api/users_resource.py [hunk 6]
```diff
@@ -1,36 +1,54 @@
...
     def post(self):
-        parser = reqparse.RequestParser()
-        parser.add_argument('username', type=str, required=True, help='Username is required')
-        parser.add_argument('password', type=str, required=True, help='Password is required')
-        parser.add_argument('isAdmin', type=bool, required=True, help='Status is required')
-        params = parser.parse_args()
-        username = params['username']
-        password = params['password']
-        isAdmin = params['isAdmin']
+        data = parser.parse_args()
```
Comment (malmike) [inline app/api/users_resource.py:23]: Validate your data before you use it

File: app/api/users_resource.py [hunk 7]
```diff
@@ -1,36 +1,54 @@
...
     def post(self):
-        pass
+        parser = reqparse.RequestParser()
+        parser.add_argument('username', type=str, required=True,
+                    help='Username is required')
+        parser.add_argument('password', type=str, required=True,
+                    help='Password is required')
+        param = parser.parse_args()
+        user = get_user_by_username(param['username'])
+
```
Comment (malmike) [inline app/api/users_resource.py:42]: Status codes?

File: app/api/users_resource.py [hunk 8]
```diff
@@ -1,36 +1,54 @@
...
 class UserLogin(Resource):
     def post(self):
-        pass
+        parser = reqparse.RequestParser()
+        parser.add_argument('username', type=str, required=True,
+                    help='Username is required')
+        parser.add_argument('password', type=str, required=True,
+                    help='Password is required')
+        param = parser.parse_args()
+        user = get_user_by_username(param['username'])
```
Comment (malmike) [inline app/api/users_resource.py:41]: Validate your data

File: app/models/menu.py [hunk 9]
```diff
@@ -1,20 +1,74 @@
-menus = []
+from datetime import datetime
+from meals import Meal, get_meal_by_id
 
 
+menus = [
```
Comment (malmike) [inline app/models/menu.py:5]: Ensure not to have hard coded data unless it is asked for

File: app/models/users.py [hunk 10]
```diff
@@ -1,4 +1,27 @@
-users = []
+users = [
```
Comment (malmike) [inline app/models/users.py:1]: Ensure not to have hard code data, unless it is asked for

**Comment by samuelbwambale** [comment]
Comments have been noted

**Pull request merged**